### ISSUE TITLE
do-while with label-continue reporting unreachable code

### ISSUE THREAD

**Issue by Wolvereness** [Microsoft/TypeScript#11880]
**TypeScript Version:**  2.0.3

**Code**

```ts
function test() {
    let foo = 0;
    testLoop: do {
        foo++;
        continue testLoop;
    } while (foo != 1);

    return foo;
}
```
[Playground](https://www.typescriptlang.org/play/index.html#src=function%20test()%20%7B%0D%0A%20%20%20%20let%20foo%20%3D%200%3B%0D%0A%20%20%20%20testLoop%3A%20do%20%7B%0D%0A%20%20%20%20%20%20%20%20foo%2B%2B%3B%0D%0A%20%20%20%20%20%20%20%20continue%20testLoop%3B%0D%0A%20%20%20%20%7D%20while%20(foo%20!%3D%201)%3B%0D%0A%0D%0A%20%20%20%20return%20foo%3B%0D%0A%7D%0D%0Adocument.body.innerHTML%20%3D%20%22%22%20%2B%20test()%3B%0D%0A)

**Expected behavior:**
No error, returns 1

**Actual behavior:**

    TS7027:Unreachable code detected.

On the return statement.

Still returns 1 on resulting code.

**Impact:**
Nested loops require labels to jump multiple levels, and this bug reports problems on code that is otherwise correct. It is avoidable (has workarounds), but it should still not erroneously report code unreachable in this situation.
